libraries/AP_Math/control.cpp

```diff
@@ -313,7 +313,7 @@ void shape_pos_vel_accel(postype_t pos_input, float vel_input, float accel_input
     vel_target += vel_input;
 
     // limit velocity between vel_min and vel_max
-    if (limit_total) {
+    if (limit_total && (is_negative(vel_min) || is_positive(vel_max))) {
```

Comment:
This test would pass if vel_min > vel_max as long as vel_max was positive or if vel_min is negative and vel_max < vel_min. I don't know what the constrain would return in that case. Perhaps you mean || to be && or alternative you need to add && vel_min < vel_max.

Reply:
` if (limit_total && (vel_max>vel_min) {`